### PR TITLE
OOP school library: set up associations 🎗️

### PR DESCRIPTION
**Pull request by ritikarawat220**
### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements

- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements

- Create a class Classroom with the following:
     - @label instance variable, should be initialized in the constructor.
     - Setter and getter for @label (remember about attr_accessor).

- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
   - Create the has-many side (a classroom has many students).
  - Create the belongs-to side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- Create a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).

- Create a class Rental with the following:
  - @date instance variable, should be initialized in the constructor.
  - Setter and getter for @date (remember about attr_accessor).

- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
  - Create the has-many side of Book and Rental (a book has many rentals).
  - Create the belongs-to side of Rental and Book (a rental belongs to a book).
  - Create the has-many side of Person and Rental (a person has many rentals).
  - Create the belongs-to side of Rental and Person (a rental belongs to a person).
  - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (elisha2kyakpo1) [inline book.rb:13]: Great job adding this function, but let's keep in mind that a book will be rented out to a person, and for this reason, we need a date and a person getting a book to be recorded.
__Hint:__
 ```
def rent(person, date)
    Rental.new(date, person, self)  # We just pass in "self" which is the "Book" itself
 end
```

File: person.rb [hunk 2]
```diff
@@ -8,21 +11,24 @@ def initialize(age, parent_permission: true, name: 'Unknown')
...
+    @rentals = []
   end
 
-  private
-
-  def of_age?
-    @age >= 18
+  def add_rental(rental)
+    @rentals << rental
   end
```
Comment (elisha2kyakpo1) [inline person.rb:19]: - [ ] We need to achieve a has-many relationship for a person with a rental.

__Hint:__
 ```
def rent(book, date)
    Rental.new(date, book, self)
 end
The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
```

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @ritikarawat220, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @ritikarawat220,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**